how-to/workspace-platform-starter/client/src/framework/menu.ts

```diff
@@ -277,3 +281,123 @@ async function getModuleMenuEntries<T extends MenuOptionType<MenuTemplateType>>(
 	}
 	return menuEntries;
 }
+
+/**
+ * Show a custom menu.
+ * @param position The position to show the menu.
+ * @param position.x The x position to show the menu.
+ * @param position.y The y position to show the menu.
+ * @param parentIdentity The identity of the parent window.
+ * @param noEntryText The text to display if there are no entries.
+ * @param menuEntries The menu entries to display.
+ * @param options The options for displaying the menu.
+ * @param options.mode Display as native menu or custom popup.
+ * @returns The menu entry.
+ */
+export async function showPopupMenu<T = unknown>(
+	position: { x: number; y: number },
+	parentIdentity: OpenFin.Identity,
+	noEntryText: string,
+	menuEntries: { label: string; customData: T; icon?: string }[],
+	options?: {
+		mode?: "native" | "custom";
+	}
+): Promise<T | undefined> {
+	if ((options?.mode ?? "native") === "native") {
+		return showNativePopupMenu(position, parentIdentity, noEntryText, menuEntries);
+	}
+
+	return showHtmlPopupMenu(position, parentIdentity, noEntryText, menuEntries);
+}
+
+/**
+ * Show a custom menu using popup window.
+ * @param position The position to show the menu.
+ * @param position.x The x position to show the menu.
+ * @param position.y The y position to show the menu.
+ * @param parentIdentity The identity of the parent window.
+ * @param noEntryText The text to display if there are no entries.
+ * @param menuEntries The menu entries to display.
+ * @returns The menu entry.
+ */
+export async function showHtmlPopupMenu<T = unknown>(
+	position: { x: number; y: number },
+	parentIdentity: OpenFin.Identity,
+	noEntryText: string,
+	menuEntries: { label: string; customData: T; icon?: string }[]
+): Promise<T | undefined> {
+	const parentWindow = fin.Window.wrapSync(parentIdentity);
+	const parentBounds = await parentWindow.getBounds();
+
+	const platformWindow = fin.Window.wrapSync(fin.me.identity);
+
+	const currentPalette = await getCurrentPalette();
+
+	const result = await platformWindow.showPopupWindow({
+		name: randomUUID(),
+		initialOptions: {
+			showTaskbarIcon: false,
+			backgroundColor: currentPalette?.backgroundPrimary,
+			customData: {
+				noEntryText,
+				menuEntries,
+				palette: {
+					backgroundPrimary: currentPalette?.backgroundPrimary,
+					textDefault: currentPalette?.textDefault,
+					inputBackground: currentPalette?.inputBackground
+				}
+			}
+		},
+		url: menuProviderOptions?.popupHtml ?? "http://localhost:8080/common/popups/menu/index.html",
```

Comment:
Are we stripping http out of the code? Maybe we need a util which is get default root where we have http://localhost once instead of spread out.

Reply:
The GH packaging scripts replace anything like that, we could use the platform provider rooturl

Reply:
Yeah for the future maybe we use platform root url and fallback to localhost: 8080 in case people start swapping out 8080 for something else and the build script misses it.